File: app/src/main/java/com/capstone/project_niyakneyak/ui/main/fragment/viewmodel/AlarmListViewModelFactory.java
package com.capstone.project_niyakneyak.ui.main.fragment.viewmodel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class AlarmListViewModelFactory implements ViewModelProvider.Factory {
    private final Application application;
    public AlarmListViewModelFactory(Application application){this.application = application;}
    @NonNull
    @Override
    @SuppressWarnings("unchecked")
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass.isAssignableFrom(AlarmListViewModel.class)) {
            return (T) new AlarmListViewModel(application);
        } else {
            throw new IllegalArgumentException("Unknown ViewModel class");
        }
    }
}

File: app/src/main/java/com/capstone/project_niyakneyak/ui/main/fragment/viewmodel/AlarmListViewModelFactory.kt
package com.capstone.project_niyakneyak.ui.main.fragment.viewmodel

import android.app.Application
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class AlarmListViewModelFactory(private val application: Application) :
    ViewModelProvider.AndroidViewModelFactory() {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return if (modelClass.isAssignableFrom(AlarmListViewModel::class.java)) {
            AlarmListViewModel(application) as T
        } else {
            throw IllegalArgumentException("Unknown ViewModel class")
        }
    }
}